modules/javafx.graphics/src/main/java/com/sun/javafx/tk/Toolkit.java

```diff
@@ -536,15 +524,7 @@ protected void notifyShutdownHooks() {
     public void notifyWindowListeners(final List<TKStage> windows) {
         for (Map.Entry<TKListener,AccessControlContext> entry : toolkitListeners.entrySet()) {
             final TKListener listener = entry.getKey();
-            final AccessControlContext acc = entry.getValue();
-            if (acc == null) {
-                throw new IllegalStateException("Invalid AccessControlContext");
-            }
-
-            AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
-                listener.changedTopLevelWindows(windows);
-                return null;
-            }, acc);
+            listener.changedTopLevelWindows(windows);
```

Comment:
Do we still need hashmap here?

Reply:
Good question: The short answer is "no", but fixing that is part of the larger cleanup of the remaining uses of AccessControlContext (along with all of the getContext methods, including the method that intersects ACC that I mentioned above).

Reply:
the loop can iterate over keySet() then

Reply:
recommendation: add a `TODO` comment with the ticket number which can be searched later.
or do you think a simple `grep AccessControlContext` should work?